cluster-api/src/main/java/com/sequenceiq/cloudbreak/cluster/status/DetailedHostStatuses.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.cloudbreak.cluster.status;
+
+import static org.apache.commons.collections4.CollectionUtils.emptyIfNull;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.sequenceiq.cloudbreak.cloud.model.HostName;
+import com.sequenceiq.cloudbreak.common.type.DetailedHealthCheck;
+import com.sequenceiq.cloudbreak.common.type.HealthCheckResult;
+import com.sequenceiq.cloudbreak.common.type.HealthCheckType;
+import com.sequenceiq.cloudbreak.common.type.HostCommissionState;
+
```

Comment:
JavaDoc. Looks like this is almost a utility class which looks at the various health checks that have been performed, and then answers some questions.
Should this, for now, be linked to faster scaling?
Something like HealthCheckInterpretationForFastScaling (or something along those lines)

Reply:
yes, I agree, since this whole change will be used for fast scaling, class names, or even java packages should represent this fact

Reply:
It'll be used initially for fast scaling. Hoping more of it will eventually be usable in other parts like the StatusCheckerJobs - especially aspects like Maintenence mode (e.g. if maint mode - ignore bad health for the node).
That, however, is a much bigger change (or tougher for me to understand anyway) - since it can have impact on cluster status and operations.

Reply:
yes, when/if the time comes when fast scaling will be default one, we can use this, and we move the whole solution into a more common place.